exporters/prometheus/exporter.go

```diff
@@ -181,6 +183,16 @@ func (c *collector) Collect(ch chan<- prometheus.Metric) {
 		ch <- c.targetInfo
 	}
 
+	var resourceAttrs []attribute.KeyValue
+	if c.resourceAttributesFilter != nil {
+		for _, kv := range metrics.Resource.Attributes() {
+			filter := *c.resourceAttributesFilter
+			if filter(kv) {
+				resourceAttrs = append(resourceAttrs, kv)
+			}
+		}
+	}
```

Comment:
This is what I would expect as a first pass.  

I would run the benchmark that already exists to see how it affects current performance, and then build a similar one to see how it does with a filter.

Reply:
i've run the benchmark before:

```
pkg: go.opentelemetry.io/otel/exporters/prometheus
BenchmarkCollect1-10              111890              9025 ns/op           35505 B/op         72 allocs/op
BenchmarkCollect10-10              48694             24922 ns/op           48411 B/op        407 allocs/op
BenchmarkCollect100-10              7225            160738 ns/op          192709 B/op       3758 allocs/op
BenchmarkCollect1000-10              679           1782206 ns/op         1566864 B/op      37118 allocs/op
BenchmarkCollect10000-10              64          16184858 ns/op        15054883 B/op     370838 allocs/op
PASS
ok      go.opentelemetry.io/otel/exporters/prometheus   6.654s
```

and here it is after the change:

```
pkg: go.opentelemetry.io/otel/exporters/prometheus
BenchmarkCollect1-10              115384              9425 ns/op           35505 B/op         72 allocs/op
BenchmarkCollect10-10              46641             24856 ns/op           48410 B/op        407 allocs/op
BenchmarkCollect100-10              7626            161435 ns/op          192703 B/op       3758 allocs/op
BenchmarkCollect1000-10              691           1720687 ns/op         1566804 B/op      37118 allocs/op
BenchmarkCollect10000-10              63          16058010 ns/op        15054975 B/op     370843 allocs/op
PASS
ok      go.opentelemetry.io/otel/exporters/prometheus   6.523s
```

Not a huge change here

Reply:
Using benchstat we see there isn't much of a difference.:

```
name             old time/op    new time/op    delta
Collect1-10        9.03µs ± 0%    9.43µs ± 0%   ~     (p=1.000 n=1+1)
Collect10-10       24.9µs ± 0%    24.9µs ± 0%   ~     (p=1.000 n=1+1)
Collect100-10       161µs ± 0%     161µs ± 0%   ~     (p=1.000 n=1+1)
Collect1000-10     1.78ms ± 0%    1.72ms ± 0%   ~     (p=1.000 n=1+1)
Collect10000-10    16.2ms ± 0%    16.1ms ± 0%   ~     (p=1.000 n=1+1)

name             old alloc/op   new alloc/op   delta
Collect1-10        35.5kB ± 0%    35.5kB ± 0%   ~     (all equal)
Collect10-10       48.4kB ± 0%    48.4kB ± 0%   ~     (p=1.000 n=1+1)
Collect100-10       193kB ± 0%     193kB ± 0%   ~     (p=1.000 n=1+1)
Collect1000-10     1.57MB ± 0%    1.57MB ± 0%   ~     (p=1.000 n=1+1)
Collect10000-10    15.1MB ± 0%    15.1MB ± 0%   ~     (p=1.000 n=1+1)

name             old allocs/op  new allocs/op  delta
Collect1-10          72.0 ± 0%      72.0 ± 0%   ~     (all equal)
Collect10-10          407 ± 0%       407 ± 0%   ~     (all equal)
Collect100-10       3.76k ± 0%     3.76k ± 0%   ~     (all equal)
Collect1000-10      37.1k ± 0%     37.1k ± 0%   ~     (all equal)
Collect10000-10      371k ± 0%      371k ± 0%   ~     (p=1.000 n=1+1)
```